f3d archlinux package do not support plugins compilation

**Describe the bug**
When installing the f3d archlinux package (maintained by @svenstaro) or f3d-git from the aur (maintainded the F3D maintainers), it is not possible to configure a plugin

**To Reproduce**
Steps to reproduce the behavior:
 1. 
 
 ```
yay -S f3d-git
cd example/plugins/example-plugin
mkdir build
cd build
cmake ../
```

 2.
 ```
 -- Configuring done (0.8s)
CMake Error at /usr/lib/cmake/vtk/VTK-targets.cmake:555 (set_target_properties):
  The link interface of target "VTK::nlohmannjson" contains:

    nlohmann_json::nlohmann_json

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  /usr/lib/cmake/vtk/vtk-config.cmake:138 (include)
  /usr/lib/cmake/f3d/f3dPlugin.cmake:157 (find_package)
  example-plugin/CMakeLists.txt:19 (f3d_plugin_build)


-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.

 ```
 

**Expected behavior**
Configure and build fine

**System Information:**
 - OS: ArchLinux


**Additional context**
This very well may be an issue with the VTK package of arch linux, but investigation is needed on our part before opening an bugreport there.

--- a/cmake/f3dPlugin.cmake
+++ b/cmake/f3dPlugin.cmake
@@ -143,7 +143,7 @@ The `NAME` argument is required. The arguments are as follows:
   * `NAME`: (Required) The name of the plugin.
   * `DESCRIPTION`: The description of the plugin.
   * `VERSION`: The version of the plugin.
-  * `VTK_MODULES`: The list of VTK modules used by the plugin to link with.
+  * `VTK_MODULES`: The list of VTK modules used by the plugin to link with. A few VTK modules are automatically linked.
   * `ADDITIONAL_RPATHS`: The list of additional RPATH for the installed binaries on Unix. VTK path is added automatically.
   * `MIMETYPE_XML_FILES`: The list of mimetype files to install. It's useful for file association on OS using Freedesktop specifications.
   * `CONFIGURATION_DIRS`: The list of configuration directories to install. Generally contain a load-plugins option and format specific options.
@@ -264,6 +264,7 @@ macro(f3d_plugin_build)
   list(TRANSFORM F3D_PLUGIN_VTK_MODULES PREPEND "VTK::")
 
   target_link_libraries(f3d-plugin-${F3D_PLUGIN_NAME} PRIVATE
+    VTK::CommonCore VTK::CommonExecutionModel VTK::IOImport
     ${F3D_PLUGIN_VTK_MODULES}
     ${modules})
 
@@ -274,7 +275,6 @@ macro(f3d_plugin_build)
       RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT plugin
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT plugin)
   endif()
-
 
   # Install configurations folders
   foreach(config_dir ${F3D_PLUGIN_CONFIGURATION_DIRS})
@@ -316,7 +316,7 @@ macro(f3d_plugin_build)
   {
     "description" : "Plugin description",
     "name" : "myPlugin",
-    "readers" : 
+    "readers" :
     [
       {
         "description" : "Reader description",

--- a/examples/plugins/example-plugin/CMakeLists.txt
+++ b/examples/plugins/example-plugin/CMakeLists.txt
@@ -20,7 +20,6 @@ f3d_plugin_build(
   NAME example
   VERSION 1.0
   DESCRIPTION "Example plugin"
-  VTK_MODULES CommonCore CommonExecutionModel
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-example-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"
   FREEDESKTOP

--- a/examples/plugins/example-plugin/module/vtk.module
+++ b/examples/plugins/example-plugin/module/vtk.module
@@ -5,4 +5,3 @@ DESCRIPTION
 DEPENDS
   VTK::CommonCore
   VTK::CommonExecutionModel
-  VTK::IOImport

--- a/plugins/alembic/CMakeLists.txt
+++ b/plugins/alembic/CMakeLists.txt
@@ -39,7 +39,7 @@ f3d_plugin_build(
   NAME alembic
   VERSION 1.0
   DESCRIPTION "Alembic support (version ${Alembic_VERSION})"
-  VTK_MODULES CommonCore CommonExecutionModel IOImport
+  VTK_MODULES CommonCore IOImport IOImage
   ADDITIONAL_RPATHS ${rpaths}
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-alembic-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"

--- a/plugins/alembic/CMakeLists.txt
+++ b/plugins/alembic/CMakeLists.txt
@@ -39,7 +39,7 @@ f3d_plugin_build(
   NAME alembic
   VERSION 1.0
   DESCRIPTION "Alembic support (version ${Alembic_VERSION})"
-  VTK_MODULES CommonCore IOImport IOImage
+  VTK_MODULES IOImage
   ADDITIONAL_RPATHS ${rpaths}
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-alembic-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"

--- a/plugins/assimp/CMakeLists.txt
+++ b/plugins/assimp/CMakeLists.txt
@@ -68,7 +68,7 @@ f3d_plugin_build(
   NAME assimp
   VERSION 1.0
   DESCRIPTION "Assimp support (version ${assimp_VERSION})"
-  VTK_MODULES CommonCore CommonExecutionModel IOImport
+  VTK_MODULES CommonCore IOImport IOImage RenderingOpenGL2
   ADDITIONAL_RPATHS ${rpaths}
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-assimp-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"

--- a/plugins/assimp/CMakeLists.txt
+++ b/plugins/assimp/CMakeLists.txt
@@ -68,7 +68,7 @@ f3d_plugin_build(
   NAME assimp
   VERSION 1.0
   DESCRIPTION "Assimp support (version ${assimp_VERSION})"
-  VTK_MODULES CommonCore IOImport IOImage RenderingOpenGL2
+  VTK_MODULES IOImage RenderingOpenGL2
   ADDITIONAL_RPATHS ${rpaths}
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-assimp-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"

--- a/plugins/draco/CMakeLists.txt
+++ b/plugins/draco/CMakeLists.txt
@@ -35,7 +35,6 @@ f3d_plugin_build(
   NAME draco
   VERSION 1.0
   DESCRIPTION "Draco support (version ${draco_VERSION})"
-  VTK_MODULES CommonCore CommonExecutionModel
   ADDITIONAL_RPATHS ${rpaths}
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-draco-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"

--- a/plugins/draco/module/vtk.module
+++ b/plugins/draco/module/vtk.module
@@ -5,7 +5,6 @@ DESCRIPTION
 DEPENDS
   VTK::CommonCore
   VTK::CommonExecutionModel
-  VTK::IOImport
   draco::draco
 TEST_DEPENDS
   VTK::TestingCore

--- a/plugins/exodus/CMakeLists.txt
+++ b/plugins/exodus/CMakeLists.txt
@@ -26,6 +26,6 @@ f3d_plugin_build(
   NAME exodus
   VERSION 1.0
   DESCRIPTION "VTK Exodus support"
-  VTK_MODULES CommonCore CommonExecutionModel IOImport IOExodus
+  VTK_MODULES CommonCore IOExodus
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"
 )

--- a/plugins/exodus/CMakeLists.txt
+++ b/plugins/exodus/CMakeLists.txt
@@ -26,6 +26,6 @@ f3d_plugin_build(
   NAME exodus
   VERSION 1.0
   DESCRIPTION "VTK Exodus support"
-  VTK_MODULES CommonCore IOExodus
+  VTK_MODULES IOExodus
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"
 )

--- a/plugins/native/CMakeLists.txt
+++ b/plugins/native/CMakeLists.txt
@@ -176,7 +176,7 @@ f3d_plugin_build(
   NAME native
   VERSION 1.0
   DESCRIPTION "Native VTK I/O support"
-  VTK_MODULES CommonCore IOImport IOXML IOParallel IOCityGML IOImage IOGeometry IOPLY
+  VTK_MODULES IOXML IOParallel IOCityGML IOImage IOGeometry IOPLY
   MIMETYPE_XML_FILES
     "${CMAKE_CURRENT_SOURCE_DIR}/f3d-3d-formats.xml"
     "${CMAKE_CURRENT_SOURCE_DIR}/f3d-3d-image-formats.xml"

--- a/plugins/native/CMakeLists.txt
+++ b/plugins/native/CMakeLists.txt
@@ -176,7 +176,7 @@ f3d_plugin_build(
   NAME native
   VERSION 1.0
   DESCRIPTION "Native VTK I/O support"
-  VTK_MODULES CommonCore CommonExecutionModel IOImport IOXML IOParallel IOCityGML IOImage IOGeometry IOPLY
+  VTK_MODULES CommonCore IOImport IOXML IOParallel IOCityGML IOImage IOGeometry IOPLY
   MIMETYPE_XML_FILES
     "${CMAKE_CURRENT_SOURCE_DIR}/f3d-3d-formats.xml"
     "${CMAKE_CURRENT_SOURCE_DIR}/f3d-3d-image-formats.xml"

--- a/plugins/occt/CMakeLists.txt
+++ b/plugins/occt/CMakeLists.txt
@@ -57,11 +57,12 @@ if (target_type STREQUAL SHARED_LIBRARY)
   list(APPEND rpaths "${OpenCASCADE_LIBRARY_DIR}")
 endif ()
 
+# RenderingOpenGL2 is needed as OpenCascade itself can expose some VTK libs that we need to link against
 f3d_plugin_build(
   NAME occt
   VERSION 1.0
   DESCRIPTION "OpenCASCADE support (version ${OpenCASCADE_VERSION})"
-  VTK_MODULES CommonCore CommonExecutionModel IOImport
+  VTK_MODULES CommonCore CommonExecutionModel FiltersGeneral RenderingOpenGL2
   ADDITIONAL_RPATHS ${rpaths}
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-occt-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"

--- a/plugins/occt/CMakeLists.txt
+++ b/plugins/occt/CMakeLists.txt
@@ -62,7 +62,7 @@ f3d_plugin_build(
   NAME occt
   VERSION 1.0
   DESCRIPTION "OpenCASCADE support (version ${OpenCASCADE_VERSION})"
-  VTK_MODULES CommonCore CommonExecutionModel FiltersGeneral RenderingOpenGL2
+  VTK_MODULES FiltersGeneral RenderingOpenGL2
   ADDITIONAL_RPATHS ${rpaths}
   MIMETYPE_XML_FILES "${CMAKE_CURRENT_SOURCE_DIR}/f3d-occt-formats.xml"
   CONFIGURATION_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/configs/config.d" "${CMAKE_CURRENT_SOURCE_DIR}/configs/thumbnail.d"

